api/python/tests/integration/data/t4_manifest.jsonl

```diff
@@ -2,4 +2,3 @@
 {"logical_key":"foo","physical_keys":["s3://my_bucket/my_data_pkg/foo"],"hash":{"type":"SHA256","value":"abcd"},"size":1024,"meta":{}}
 {"logical_key":"bar.csv","physical_keys":["s3://my_bucket/my_data_pkg/bar.csv"],"hash":{"type":"SHA256","value":"abcd"},"size":1024,"meta":{}}
 {"logical_key":"baz/bat","physical_keys":["s3://my_bucket/my_data_pkg/baz/bat"],"hash":{"type":"SHA256","value":"abcd"},"size":1024,"meta":{}}
-{"logical_key":"my_data_pkg/data_frames/","physical_keys":["s3://my_bucket/my_data_pkg/data_frames/"],"hash":{"type":"SHA256","value":"abcd"},"size":1024,"meta":{}}
```

Comment:
any reason there were deleted in this PR?

Reply:
They end with `/` - but directories don't make sense as keys, right? The existing code somehow ignored that - but I've added sanity checks everywhere in the new code, so the tests started failing.

Reply:
Ah, cool. We do allow directory level logical keys, but they should not have physical keys (they are used for storing directory level metadata)